auth_manager.go

```diff
@@ -89,6 +103,56 @@ func (b *DefaultSessionManager) Init(store storage.Handler) {
 	b.disableCacheSessionState = config.Global().LocalSessionCache.DisableCacheSessionState
 	b.store = store
 	b.store.Connect()
+
+	if b.asyncWrites {
+		//check pool size in config and set to 50 if unset
+		b.poolSize = config.Global().SessionUpdatePoolSize
+		if b.poolSize <= 0 {
+			b.poolSize = 50
+		}
+
+		log.WithField("Auth Manager", b.poolSize).Debug("Session update async pool size")
+
+		b.updateChan = make(chan *SessionUpdate)
+
+		//start worker pool
+		atomic.SwapUint32(&b.shouldStop, 0)
+		for i := 0; i < b.poolSize; i++ {
+			b.poolWG.Add(1)
+			go b.updateWorker()
+		}
+	}
+}
+
+func (b *DefaultSessionManager) updateWorker() {
+	defer b.poolWG.Done()
+
+	for {
```

Comment:
here probably here would be better to use `range` cycle instead of `select` as we don't have to read from several channels, i.e.:
```
for u := range b.updateChan {
    ....
}
``` 
cycle will exit automatically once channel is closed